### ISSUE TITLE
`void_checks` should have tests with assignment patterns

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#4179]
Child issue of https://github.com/dart-lang/linter/issues/3624

I think there should be a test case like:

```dart
void f(void p) {
  (p, ) = (7, );
}
```

in which the lint rule would report that assigning to `p` violates `void_checks`.

Also maybe something with object patterns? like `class C<T> { T f; }` and `case C<void>(f: var x): print('hi');`...

**Comment by pq** [comment]
Nice catch!
